static/app/components/events/interfaces/spans/traceView.tsx

```diff
@@ -28,68 +24,85 @@ class TraceView extends PureComponent<Props> {
+    if (!getTraceContext(waterfallModel.event)) {
       return (
         <EmptyStateWarning>
           <p>{t('There is no trace for this transaction')}</p>
         </EmptyStateWarning>
       );
     }
 
-    const {organization, waterfallModel} = this.props;
-
     return (
       <DragManager interactiveLayerRef={this.minimapInteractiveRef}>
         {(dragProps: DragManagerChildrenProps) => (
-          <CursorGuideHandler.Provider
-            interactiveLayerRef={this.minimapInteractiveRef}
-            dragProps={dragProps}
-            trace={parsedTrace}
-          >
-            <DividerHandlerManager.Provider interactiveLayerRef={this.traceViewRef}>
-              <DividerHandlerManager.Consumer>
-                {dividerHandlerChildrenProps => {
-                  return (
-                    <ScrollbarManager.Provider
-                      dividerPosition={dividerHandlerChildrenProps.dividerPosition}
-                      interactiveLayerRef={this.virtualScrollBarContainerRef}
-                      dragProps={dragProps}
-                    >
-                      {this.renderHeader(dragProps, parsedTrace)}
-                      <Observer>
-                        {() => (
-                          <SpanTree
-                            traceViewRef={this.traceViewRef}
-                            event={event}
-                            trace={parsedTrace}
+          <Observer>
+            {() => {
+              const parsedTrace = waterfallModel.parsedTrace;
+              return (
+                <CursorGuideHandler.Provider
+                  interactiveLayerRef={this.minimapInteractiveRef}
+                  dragProps={dragProps}
+                  trace={parsedTrace}
+                >
+                  <DividerHandlerManager.Provider interactiveLayerRef={this.traceViewRef}>
+                    <DividerHandlerManager.Consumer>
+                      {dividerHandlerChildrenProps => {
+                        return (
+                          <ScrollbarManager.Provider
+                            dividerPosition={dividerHandlerChildrenProps.dividerPosition}
+                            interactiveLayerRef={this.virtualScrollBarContainerRef}
                             dragProps={dragProps}
-                            filterSpans={waterfallModel.filterSpans}
-                            organization={organization}
-                            operationNameFilters={waterfallModel.operationNameFilters}
-                          />
-                        )}
-                      </Observer>
-                    </ScrollbarManager.Provider>
-                  );
-                }}
-              </DividerHandlerManager.Consumer>
-            </DividerHandlerManager.Provider>
-          </CursorGuideHandler.Provider>
+                          >
+                            {this.renderHeader(dragProps)}
+                            <Observer>
+                              {() => {
```

Comment:
There is a reason for the split. The gist is that the header doesn't need to be re-rendered if the properties it depends on  within `waterfallModel` does not change. This can be demonstrated if a fuzzy search is performed. The span tree `<Observer>` region will be re-rendered, but the header will not be re-rendered. 

This is generally how mobx works by using Proxies to see which properties were accessed on the initial render.